Project/app/src/main/java/org/ionproject/android/class_section/ClassSectionDto.kt

```diff
@@ -0,0 +1,27 @@
+package org.ionproject.android.class_section
+
+import org.ionproject.android.common.model.ClassSection
+import org.ionproject.android.common.siren.EmbeddedEntity
+import org.ionproject.android.common.siren.MappingFromSirenException
+import org.ionproject.android.common.siren.SirenEntity
+import java.net.URI
+
+/**
+ * Extension method to convert from [SirenEntity] to [ClassSection]
+ */
+fun SirenEntity.toClassSection(): ClassSection {
+    val calendarURI: URI? = (entities?.first() as EmbeddedEntity).links?.first()?.href
+
+    if (properties != null && properties["course"] != null && properties["class"] != null
+        && properties["id"] != null
+    ) {
+        //Using double bang operator because we are sure this properties cannot be null here
+        return ClassSection(
+            course = properties["course"]!!,
```

Comment:
Again... Please do not use !!
Use properties.getOrDefault() or something else. If your intention is to force an exception to be thrown, then create the appropriate exception and throw it.
BTW, you state that "we are sure that it cannot be null". Are you really?

Reply:
This has been changed in ClassesSelection branch.